replica/database.cc

```diff
@@ -1013,31 +1013,46 @@ void database::remove(const table& cf) noexcept {
     }
 }
 
-future<> database::drop_column_family(const sstring& ks_name, const sstring& cf_name, timestamp_func tsf, bool snapshot) {
-    auto& ks = find_keyspace(ks_name);
-    auto uuid = find_uuid(ks_name, cf_name);
-    lw_shared_ptr<table> cf;
-    try {
-        cf = _column_families.at(uuid);
-        drop_repair_history_map_for_table(uuid);
-    } catch (std::out_of_range&) {
-        on_internal_error(dblog, fmt::format("drop_column_family {}.{}: UUID={} not found", ks_name, cf_name, uuid));
-    }
-    dblog.debug("Dropping {}.{}", ks_name, cf_name);
-    remove(*cf);
-    cf->clear_views();
-    co_await cf->await_pending_ops();
-    co_await _querier_cache.evict_all_for_table(cf->schema()->id());
-    auto f = co_await coroutine::as_future(truncate(ks, *cf, std::move(tsf), snapshot));
-    co_await cf->stop();
-    f.get(); // re-throw exception from truncate() if any
+future<> database::drop_column_family(table& cf) {
+    auto uuid = cf.schema()->id();
+    drop_repair_history_map_for_table(uuid);
+    remove(cf);
+    cf.clear_views();
+    co_await cf.await_pending_ops();
+    co_await _querier_cache.evict_all_for_table(uuid);
+}
+
+future<std::vector<foreign_ptr<lw_shared_ptr<table>>>> database::get_table_on_all_shards(sharded<database>& sharded_db, utils::UUID uuid) {
+    std::vector<foreign_ptr<lw_shared_ptr<table>>> table_shards;
```

Comment:
perhaps we can make foreign_ptr<lw_shared_ptr<table>>> easier to read with:

using foreign_table_ptr = make foreign_ptr<lw_shared_ptr<table>>>

Reply:
yeah.  BTW, I once had a proposal to define short names for `foreign_unique_ptr`, `foreign_lw_shared_ptr`, `foreign_shared_ptr`, etc.  but it didn't go through.